### ISSUE TITLE
Using NuGet repository with password encryption - Not Supported on non-Windows Platforms

### ISSUE THREAD

**Issue by livarcocc** [NuGet/Home#5526]
_From @aleslash on July 3, 2017 17:48_

## Steps to reproduce
Use local nuget repository that demands password encription
My nuget.config works fine in Windows environment, but on linux OS it doesn't.

## Expected  behavior
Successfully connect with local Nuget repository.

## Actual behavior
  Restoring packages for /opt/apl/testeDotnet/testeDotnet.csproj...
  Retrying 'FindPackagesByIdAsyncCore' for source 'http://<removed>/api/nuget/nuget-remote/FindPackagesById()?id='Microsoft.NETCore.App''.
  Password decryption is not supported on .NET Core for this platform. The following feed uses an encrypted password: 'Artifactory'. You can use a clear text password as a workaround.
    Encryption is not supported on non-Windows platforms.
  Retrying 'FindPackagesByIdAsyncCore' for source 'http:// <removed>/api/nuget/nuget-remote/FindPackagesById()?id='Microsoft.NETCore.App''.
  Password decryption is not supported on .NET Core for this platform. The following feed uses an encrypted password: 'Artifactory'. You can use a clear text password as a workaround.
    Encryption is not supported on non-Windows platforms.
/opt/apl/dotnet/sdk/1.0.4/NuGet.targets(97,5): error : Failed to retrieve information about 'Microsoft.NETCore.App' from remote source 'http:// <removed>/api/nuget/nuget-remote/FindPackagesById()?id='Microsoft.NETCore.App''. [/opt/apl/testeDotnet/testeDotnet.csproj]
/opt/apl/dotnet/sdk/1.0.4/NuGet.targets(97,5): error :   Password decryption is not supported on .NET Core for this platform. The following feed uses an encrypted password: 'Artifactory'. You can use a clear text password as a workaround. [/opt/apl/testeDotnet/testeDotnet.csproj]
/opt/apl/dotnet/sdk/1.0.4/NuGet.targets(97,5): error :   Encryption is not supported on non-Windows platforms. [/opt/apl/testeDotnet/testeDotnet.csproj]

## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.4)

Product Information:
 Version:            1.0.4
 Commit SHA-1 hash:  af1e6684fd

Runtime Environment:
 OS Name:     rhel
 OS Version:  7.2
 OS Platform: Linux
 RID:         rhel.7.2-x64
 Base Path:   /opt/apl/dotnet/sdk/1.0.4

In our case the IT Security Policy cannot be changed, so we must use encripted password.

_Copied from original issue: dotnet/cli#7062_

**Comment by nkolev92** [comment]
Duplicate of https://github.com/NuGet/Home/issues/1851

**Comment by LoicGombeaud** [comment]
@nkolev92 How is this a duplicate of #5447?
#5447 is an issue related to the Windows platform, whereas this issue is explicitly about non-Windows platform

**Comment by nkolev92** [comment]
#5447 is about support for cross platform auth through dotnet.exe. 
It's the solution to the scenario at hand raised by the OP. 

There's an issue for the encryption of passwords cross platform https://github.com/NuGet/Home/issues/1851, but that's not blocked on NuGet.

**Comment by dhurimkelmendi** [comment]
I solved this by using Visual Studio for Mac to add the source. Solution/Project -> Manage Nuget repositories -> Add source.

**Comment by p10tyr** [comment]
What is the solution for this on Linux? 
I don't understand how DotNet has been adopted to run on Linux.. but has no credential provider made for Linux?
The advice to strore PAT in plain text .. 4 years later .. sounds like incredibly bad advice?
